applications/OptimizationApplication/custom_utilities/optimization_utils.cpp

```diff
@@ -0,0 +1,150 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   license: OptimizationApplication/license.txt
+//
+//  Main author:     Suneth Warnakulasuriya
+//
+
+// System includes
+#include <cmath>
+
+// Project includes
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+#include "utilities/variable_utils.h"
+
+// Application includes
+
+// Include base h
+#include "optimization_utils.h"
+
+namespace Kratos
+{
+template<class TContainerType>
+GeometryData::KratosGeometryType OptimizationUtils::GetContainerEntityGeometryType(
+    const TContainerType& rContainer,
+    const DataCommunicator& rDataCommunicator)
+{
+    KRATOS_TRY
+
+    if (rContainer.size() > 0) {
+        const auto first_geometry_type = rContainer.begin()->GetGeometry().GetGeometryType();
+        const bool local_value = block_for_each<MinReduction<bool>>(rContainer, [&](const auto& rEntity) -> IndexType {
+            return first_geometry_type == rEntity.GetGeometry().GetGeometryType();
+        });
+
+        if (rDataCommunicator.AndReduceAll(local_value)) {
+            return first_geometry_type;
+        } else {
+            return GeometryData::KratosGeometryType::Kratos_generic_type;
+        }
+    } else {
+        return GeometryData::KratosGeometryType::Kratos_generic_type;
+    }
+
+    KRATOS_CATCH("")
+}
+
+template<class TContainerType, class TDataType>
+bool OptimizationUtils::IsVariableExistsInAllContainerProperties(
```

Comment:
```suggestion
bool OptimizationUtils::DoesVariableExistInAllContainerProperties(
```
or "VariableExistInAllContainerProperties()"
I was a little bit confused by the is... in front and find it more readable this way.
Is there a convention in Kratos to only use Is... if the function returns a bool?

Reply:
`Is` is used in Kratos to let the developer know that it is returining a bool.